drivers/modem/ublox-sara-n310.c

```diff
@@ -0,0 +1,1431 @@
+/*
+ * Copyright (c) 2022 Abel Sensors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ublox_sara_n310
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(modem_ublox_sara_n310, CONFIG_MODEM_LOG_LEVEL);
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+
+#include <net/net_if.h>
+#include <net/net_offload.h>
+#include <net/socket_offload.h>
+
+#include <kernel.h>
+#include <errno.h>
+#include <zephyr.h>
+
+#include <zephyr/drivers/modem/ublox-sara-n310.h>
+#include <zephyr/drivers/gpio.h>
+
+#include "modem_context.h"
+#include "modem_iface_uart.h"
+#include "modem_cmd_handler.h"
+#include "modem_socket.h"
+
```

Comment:
Use node references instead of names.  See how uses `DEVICE_DT_GET` instead of device_get_binding and utilizes devicetree node references.

Reply:
Just to be sure that im interpreting this correctly. So we should use `DT_NODELABEL`? For getting a reference to the correct node (I know `DT_INST` should no be used).

Reply:
When I use the following code:
```
#define N310_NODE DT_INST(0, ublox_sara_n310)
#define N310_DEV DEVICE_DT_GET(N310_NODE)
```
I get a 19 error (No such device). @galak